### PR TITLE
Move some Drive state to zustand

### PR DESCRIPTION
**Pull request by somebody1234**
### Pull Request Description
- Move Asset Panel open state to zustand
  - This avoids re-rendering the entire Data Catalog view every time an asset is clicked
- Move Asset Panel props to zustand
- Move search suggestions to zustand
- Memoize `AssetRow` to avoid re-rendering every row when unnecessary

### Important Notes
It's not a perfect solution to the performance issues, however selection performance seems to have improved quite nicely.

### Testing Instructions
- Click a row in the Assets Table and use the arrow keys to move - this is the fastest way to trigger selection changes.
  - Compare performance with `develop` of course.
  - The performance degradation on `develop` is more noticeable with many assets (e.g. 100+ or 200+)

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] ~~The documentation has been updated, if necessary.~~
- [x] ~~Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.~~
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] ~~Unit tests have been written where possible.~~

### CODE REVIEW

**Comment by PabloBuchu** [comment]
1. Something is really broken here I think. Can't close it with error shown on screenshot. Is this because I had it opened before closing on different branch?
Screenshot 2024-09-03 at 13 17 55

**Comment by somebody1234** [comment]
yeah my bad, idk what happened there... will look into asap since it should be a relatively easy fix

**Comment by somebody1234** [comment]
found the issue, pushing rn

**Comment by somebody1234** [comment]
status update: will be continuing to investigate perf issues here.

**Comment by somebody1234** [comment]
pinging @MrFlashAccount for CR.

File: app/dashboard/src/hooks/eventCallbackHooks.ts [hunk 1]
```diff
@@ -1,23 +1,19 @@
...
-import * as React from 'react'
+/** @file `useEventCallback` shim */
+import { useCallback } from 'react'
 
-import * as syncRef from '#/hooks/syncRefHooks'
+import { useSyncRef } from '#/hooks/syncRefHooks'
 
 /**
- * useEvent shim.
+ * `useEventCallback` shim.
```
Comment (MrFlashAccount) [inline app/dashboard/src/hooks/eventCallbackHooks.ts]: it's a `useEvent` shim, and _should_ be called `useEvent`, but `useEvent` already exist in the project 😊

**Review by MrFlashAccount — APPROVED** [review]

**Comment by somebody1234** [comment]
uh oh...

**Pull request merged**